Disable linking to GOTM

dazlich opened the pull request:
Because of license restrictions linking to GOTM is disabled. The file calling GOTM routines has been renamed to avoid referencing GOTM and calls and references to GOTM-related coded have been commented.

dazlich committed:
Disable linking to GOTM

Because of license restrictions linking to GOTM is disabled.
The file calling GOTM routines has been renamed to avoid
referencing GOTM and calls and references to GOTM-related coded
have been commented.

sjsprecious commented on src/mode_forward/mpas_ocn_forward_mode.F:
```diff
@@ -90,7 +90,7 @@ module ocn_forward_mode
 
    use ocn_vmix
-   use ocn_vmix_gotm, only: ocn_vmix_gotm_finalize
+   !use ocn_vmix_gotm, only: ocn_vmix_gotm_finalize
```
Do you plan to use it again once the GOTM license issue is addressed? If not, maybe we just delete this line for cleanliness?

sjsprecious commented on src/mode_forward/mpas_ocn_forward_mode.F:
```diff
@@ -876,7 +876,7 @@ function ocn_forward_mode_finalize(domain) result(iErr)!{{{
       call ocn_analysis_finalize(domain, ierr)
 
-      call ocn_vmix_gotm_finalize(domain, ierr)
+      !call ocn_vmix_gotm_finalize(domain, ierr)
```
the same as above.

sjsprecious commented on src/shared/mpas_ocn_vmix.F:
```diff
@@ -31,7 +31,7 @@ module ocn_vmix
    use ocn_config
    use ocn_vmix_cvmix
-   use ocn_vmix_gotm
+   !use ocn_vmix_gotm
```
the same as above.

sjsprecious commented on src/shared/mpas_ocn_vmix.F:
```diff
@@ -189,7 +189,7 @@ subroutine ocn_vmix_coefs(meshPool, statePool, forcingPool, scratchPool, err, ti
       call ocn_vmix_coefs_cvmix_build(meshPool, statePool, forcingPool, err1, timeLevel)
       call ocn_vmix_coefs_redi_build(meshPool, statePool, err2, timeLevel)
-      call ocn_vmix_coefs_gotm_build(statePool, forcingPool, err3, timeLevel)
+      !call ocn_vmix_coefs_gotm_build(statePool, forcingPool, err3, timeLevel)
```
the same as above.

sjsprecious commented on src/shared/mpas_ocn_vmix.F:
```diff
@@ -1570,8 +1570,8 @@ subroutine ocn_vmix_init(domain, err)!{{{
       err = ior(err, err_tmp)
-      call ocn_vmix_gotm_init(domain,err_tmp)
-      err = ior(err, err_tmp)
+      !call ocn_vmix_gotm_init(domain,err_tmp)
+      !err = ior(err, err_tmp)
```
the same as above.

sjsprecious commented on cime_config/buildlib:
```diff
@@ -41,7 +41,6 @@ def _copy_files_to_blddir(case):
                 os.path.join(objroot, "ocn", "source"),
                 os.path.join(objroot, "ocn", "input"),
                 os.path.join(objroot, "ocn", "BGC"),
-                os.path.join(objroot, "ocn", "gotm"),
```
Can you confirm that after removing GOTM, MPAS-O still builds and runs?

sjsprecious reviewed (commented):
Thanks @dazlich for working on this revision. I just have a few clarification questions before approving it.

Pranay-Reddy-Kommera commented:
Thank you @dazlich for working on making GOTM optional. 
I agree with @sjsprecious. We can remove any 'GOTM' related calls if we are not going to use it in the EarthWorks execution configurations.

dazlich commented:
@Pranay-Reddy-Kommera @sjsprecious additionally, I think I should remove the config_use_gotm switch from the code.

Pranay-Reddy-Kommera commented:
I agree @dazlich. 
There are additional GOTM-related parameters in various namelist files within mpas-ocean/bld (see: https://github.com/EarthWorksOrg/mpas-ocean/tree/1438145845e9c8d17dfad2404e3be75f7dbcf673/bld). It is unclear for me whether these parameters are currently being utilized in the runs; if not, they can be considered for removal.

dazlich committed:
Remove GOTM from mpas-ocean

Along with no longer linking to GOTM, this commit removes all
trace of GOTM from mpas-ocean.

dazlich commented:
@sjsprecious @Pranay-Reddy-Kommera - Ok, I've made a wholesale removal of GOTM from mpas-ocean. It builds and runs.

sjsprecious reviewed (approved):
Thanks @dazlich . LGTM.

Pranay-Reddy-Kommera commented:
Thank you @dazlich 
Changes look good to me.
